### PR TITLE
Hosting Command Palette: Add a few account commands 

### PR DESCRIPTION
**Pull request by katinthehatsite**
Closes https://github.com/Automattic/dotcom-forge/issues/4829

## Proposed Changes

This PR adds three new account commands:

* `Change plan on a site` - this command navigates to `/plans/{yourSite}`
* `Manage plan on a site` - this command navigates to `/plans/my-plan/{yourSite}`

Screenshot 2023-12-11 at 4 33 44 PM

For these two commands, I have set the credit card as an icon but am happy to change it to something else. In my opinion, I am not sure if we need both commands - I feel like `Change plan on a site` could be useful as it brings you to the overview of all the plans and this is where we can encourage the customer to upgrade. However, `/plans/my-plan/` (`Manage my plan`) page does not seem to be too useful for me - I don't know how often I would navigate to that page offer to see the overview of all the features that come with my plan. 

* `Get help` - this command opens a modal for the help center where the customers can search for questions

Screenshot 2023-12-11 at 4 32 50 PM

The position of the modal is what it is on other pages - not sure if we need to adjust it a bit more aligned. If so, we might need to discuss what would be a better position.

## Testing Instructions

* Navigate to Calypso Live Link
* Navigate to `sites` page
* Press `cmd + k` or `ctrl + k` to trigger the command palette
* Search for `Change plan on a site` command
* Select a site to change plan
* Confirm you are being brought to  `/plans/{yourSite}`
---
* Navigate to `sites` page
* Press `cmd + k` or `ctrl + k` to trigger the command palette
* Search for `Manage plan on a site` command
* Select a site to change plan
* Confirm you are being brought to  `/plans/my-plan/{yourSite}`
---
* Navigate to `sites` page
* Press `cmd + k` or `ctrl + k` to trigger the command palette
* Search for `Get help`
* Click on the command and confirm that the help modal appears

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

### CODE REVIEW

File: client/sites-dashboard/components/wpcom-smp-commands.tsx [hunk 1]
```diff
@@ -983,6 +997,43 @@ export const useCommandsArrayWpcom = ( {
 			},
 			icon: pluginsIcon,
 		},
+		{
+			name: 'changePlan',
+			label: __( 'Change plan on a site' ),
```
Comment (danielbachhuber) [inline client/sites-dashboard/components/wpcom-smp-commands.tsx]:
```suggestion
			label: __( 'Change site plan' ),
```

I think 'Change plan on a site' is to close to a full sentence. The other commands are more focused on statements.

File: client/sites-dashboard/components/wpcom-smp-commands.tsx [hunk 2]
```diff
@@ -983,6 +997,43 @@ export const useCommandsArrayWpcom = ( {
...
+					close();
+					navigate( `/plans/${ site.slug }` );
+				},
+				filter: ( site: SiteExcerptData ) => ! isP2Site( site ) && ! site?.is_wpcom_staging_site,
+			},
+			icon: creditCardIcon,
+		},
+		{
+			name: 'manageMyPlan',
+			label: __( 'Manage plan on a site' ),
```
Comment (danielbachhuber) [inline client/sites-dashboard/components/wpcom-smp-commands.tsx]:
```suggestion
			label: __( 'Manage site plan' ),
```

**Review by danielbachhuber — CHANGES_REQUESTED** [review]

File: client/sites-dashboard/components/wpcom-smp-commands.tsx (hunk 1 above)
Comment (katinthehatsite) [reply client/sites-dashboard/components/wpcom-smp-commands.tsx]: Changed in https://github.com/Automattic/wp-calypso/pull/85083/commits/47e470467aa9010a7f2697c44147d98d69aef2d4

File: client/sites-dashboard/components/wpcom-smp-commands.tsx (hunk 2 above)
Comment (katinthehatsite) [reply client/sites-dashboard/components/wpcom-smp-commands.tsx]: Changed in https://github.com/Automattic/wp-calypso/pull/85083/commits/47e470467aa9010a7f2697c44147d98d69aef2d4

File: client/sites-dashboard/components/wpcom-smp-commands.tsx [hunk 3]
```diff
@@ -983,6 +997,43 @@ export const useCommandsArrayWpcom = ( {
...
+		{
+			name: 'changePlan',
+			label: __( 'Change site plan' ),
+			searchLabel: [
+				_x( 'upgrade plan', 'Keyword for the Change site plan command' ),
+				_x( 'change plan', 'Keyword for the Change site plan command' ),
+				_x( 'add plan', 'Keyword for the Change site plan command' ),
+			].join( ' ' ),
+			context: [ '/sites' ],
+			callback: setStateCallback( 'changePlan', __( 'Select site to change plan' ) ),
```
Comment (sejas) [inline client/sites-dashboard/components/wpcom-smp-commands.tsx:1012]: thanks for adding the placeholder text 👌

**Review by sejas — APPROVED** [review]
Thanks for changing the labels.
It works as expected.
A nice to have would be autofocus the search text input in the help modal, but [inspecting the code](https://github.com/Automattic/wp-calypso/blob/854d895269e7903f64fcecbfb264b2bea0ec4c41/packages/data-stores/src/help-center/actions.ts#L75) seems to be more complex than it should be for this PR. We can observe the metrics and decide to implement it later if people use that command.

## Screencast
I also tried opening multiple times the "get help" command. Working as expected.

https://github.com/Automattic/wp-calypso/assets/779993/1db68fb8-3172-4db4-8975-1c32a0a4ef66

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/10265135

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @katinthehatsite for including a screenshot in the description! This is really helpful for our translators.

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/10265135).

**Pull request merged**